### PR TITLE
Log DNS queries made in our VPC

### PR DESCRIPTION
**Pull request by jimleroyer**
# Summary | Résumé

* Created a new log group to store the DNS requests made within our VPC.
* Created a Resolver query logging within Route 53 to log DNS requests.

# Test instructions | Instructions pour tester la modification

I tried to tests first with CLI but I didn't have access to latest documented commands despite having latest available (at least from Brew), such as `create-resolver-query-log-config` and `associate-resolver-query-log-config`.

For now, we'll have to test by applying Terraform changes when merging into the `main` branch.

# Reviewer checklist | Liste de vérification du réviseur

This is a suggested checklist of questions reviewers might ask during their
review | Voici une suggestion de liste de vérification comprenant des questions
que les réviseurs pourraient poser pendant leur examen :

- [ ] Does this meet a user need? | Est-ce que ça répond à un besoin utilisateur?
- [ ] Is it accessible? | Est-ce que c’est accessible?
- [ ] Is it translated between both offical languages? | Est-ce dans les deux
      langues officielles?
- [ ] Is the code maintainable? | Est-ce que le code peut être maintenu?
- [ ] Have you tested it? | L’avez-vous testé?
- [ ] Are there automated tests? | Y a-t-il des tests automatisés?
- [ ] Does this cause automated test coverage to drop? | Est-ce que ça entraîne
      une baisse de la quantité de code couvert par les tests automatisés?
- [ ] Does this break existing functionality? | Est-ce que ça brise une
      fonctionnalité existante?
- [ ] Should this be split into smaller PRs to decrease change risk? | Est-ce
      que ça devrait être divisé en de plus petites demandes de tirage (« pull
      requests ») afin de réduire le risque lié aux modifications?
- [ ] Does this change the privacy policy? | Est-ce que ça entraîne une
      modification de la politique de confidentialité?
- [ ] Does this introduce any security concerns? | Est-ce que ça introduit des
      préoccupations liées à la sécurité?
- [ ] Does this significantly alter performance? | Est-ce que ça modifie de
      façon importante la performance?
- [ ] What is the risk level of using added dependencies? | Quel est le degré de
      risque d’utiliser des dépendances ajoutées?
- [ ] Should any documentation be updated as a result of this? (i.e. README
      setup, etc.) | Faudra-t-il mettre à jour la documentation à la suite de ce
      changement (fichier README, etc.)?

### CODE REVIEW

File: aws/common/route53.tf [hunk 1]
```diff
@@ -0,0 +1,12 @@
+###
+# AWS Route 53 for Notification application
+###
+
+resource "aws_route53_resolver_query_log_config" "dns_query_log_config" {
+  name            = "us-east-1/${var.account_id}/dns_query_log_config"
+  destination_arn = [aws_cloudwatch_log_group.route53_resolver_query_log.arn]
```
Comment (AntoineAugusti) [inline aws/common/route53.tf:7]: This should be a string

File: aws/common/cloudwatch_log.tf [hunk 2]
```diff
@@ -49,6 +49,18 @@ resource "aws_cloudwatch_log_group" "sns_deliveries_failures_us_west_2" {
   }
 }
 
+resource "aws_cloudwatch_log_group" "route53_resolver_query_log" {
+  provider = aws.us-east-1
```
Comment (AntoineAugusti) [inline aws/common/cloudwatch_log.tf:53]: I think that you read the wrong documentation page. [This page](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/query-logs.html) is about query logging when you use Route53 as a DNS resolver for your own domain that you're managing on AWS.

This is not what we're using, we're using their resolver for everything in our VPC.

You can visit https://ca-central-1.console.aws.amazon.com/route53resolver/home?region=ca-central-1#/query-logging in our staging account to see that we could be able to use a log group in ca-central-1

File: aws/common/route53.tf [hunk 3]
```diff
@@ -0,0 +1,12 @@
+###
+# AWS Route 53 for Notification application
+###
+
+resource "aws_route53_resolver_query_log_config" "dns_query_log_config" {
```
Comment (AntoineAugusti) [inline aws/common/route53.tf:5]: Needs also a https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_resolver_query_log_config_association to associate this with our VPC!

File: aws/common/route53.tf [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  name            = "us-east-1/${var.account_id}/dns_query_log_config"
```
Comment (AntoineAugusti) [inline aws/common/route53.tf:6]: Change the name as this will not be in this region. You can use `var.region`

**Review by AntoineAugusti — CHANGES_REQUESTED** [review]

File: aws/common/route53.tf (hunk 1 above)
Comment (jimleroyer) [reply aws/common/route53.tf:7]: I did notice from the same message given by the build. Other examples in the code does not enclose in strings though for similarly ARN references. Should we simply enclose it in double quotes and that will be ok?

Comment (AntoineAugusti) [reply aws/common/route53.tf:7]: I think you don't need to enclose it in double quotes! `make fmt` would convert that as needed

File: aws/common/cloudwatch_log.tf (hunk 2 above)
Comment (jimleroyer) [reply aws/common/cloudwatch_log.tf:53]: Sorry, I don't understand your comment. A log group is defined here to be used later by the aws route 53 resolver query log. The link you referred to is to create a Route 53 resolver query log too. 

Let's speak on Slack about this.

Comment (jimleroyer) [reply aws/common/cloudwatch_log.tf]: Spoke with Antoine offline. He meant to say that the provider's region is incorrect.

File: aws/common/route53.tf (hunk 4 above)
Comment (jimleroyer) [reply aws/common/route53.tf]: I was 😫 🍸

**Review by AntoineAugusti — APPROVED** [review]
:shipit: 

just after you fix the name of the resolver query log config

**Comment by jimleroyer** [comment]
> :shipit:
> 
> just after you fix the name of the resolver query log config

The name was changed to `name            = "${var.region}/${var.account_id}/dns_query_log_config"` in my last commit. Do you suggest something else?

**Pull request merged**